src/briefcase/integrations/subprocess.py

```diff
@@ -608,17 +612,16 @@ def _stream_output_thread(self, popen_process, filter_func):
         # ValueError is raised if stdout is unexpectedly closed.
         # This can happen if the user starts spamming CTRL+C, for instance.
         # Silently exit to avoid Python printing the exception to the console.
-        # StopIteration can be raised by a filter function to indicate that
+        # StopStreaming can be raised by a filter function to indicate that
         # there's no need to stream any more.
-        with suppress(ValueError, StopIteration):
+        with suppress(ValueError, StopStreaming):
```

Comment:
There's quite a large amount of code being run inside this `suppress` block. So to avoid hiding `ValueErrors` that are raised for other reasons, it would be safer to remove the suppression of `ValueError`, and instead put an exception handler around `readline` alone, which catches `ValueError` and raises `StopStreaming`.

Reply:
That's a good point. Furthermore, the output streamer never implemented any robust exception handling since it was pretty simple; however, any exceptions will simply spill out in to the console as the thread rolls up. We may want to consider handling that at least involves including such exceptions in the log file....perhaps via `Log.add_log_file_extra()`.

Reply:
I've restructured the streaming function to have more specific exception handling, and added exception handling for safety while streaming.